### PR TITLE
feat: implement waf check framework and include starter

### PR DESCRIPTION
**Pull request by matt-FFFFFF**
Implements a way to check attribute values are correct.

Added WAF (well architected framework) package to surface and make easily extensible.

### CODE REVIEW

File: waf/azurermStorageAccount.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package waf
+
+import "github.com/Azure/tflint-ruleset-avm/attrvalue"
+
+// AzurermStorageAccountAccountReplicationType checks whether the account_replication_type is set to ZRS.
+var AzurermStorageAccountAccountReplicationType = func() *attrvalue.StringRule {
+	return attrvalue.NewStringRule(
+		"azurerm_storage_account",
+		"account_replication_type",
+		[]string{"ZRS"},
```
Comment (jtracey93) [inline waf/azurermStorageAccount.go:10]: Should we also include GA-ZRS?

**Review by jtracey93 — CHANGES_REQUESTED** [review]

File: waf/azurermStorageAccount.go (hunk 1 above)
Comment (matt-FFFFFF) [reply waf/azurermStorageAccount.go:10]: I think this depends on what we want AVM to be.

We state that we should be HA by default but do we state DR?

Remember this is just the default value provided by the module.

**Comment by matt-FFFFFF** [comment]
Do not merge until all tests pass..

https://github.com/terraform-linters/tflint/discussions/1996

File: waf/azurermStorageAccount.go (hunk 1 above)
Comment (jtracey93) [reply waf/azurermStorageAccount.go:10]: yeah thats fair, I was thinking that an OR for `ZRS` or `GZRS` would satisfy the test as one is just an enhancement on the other

**Comment by matt-FFFFFF** [comment]
> Do not merge until all tests pass..
> 
> [terraform-linters/tflint#1996](https://github.com/terraform-linters/tflint/discussions/1996)

This is resolved now - limitation of the tflint test runner

**Pull request closed without merging**